source/docs/hardware/control/vmx-pi.rst

```diff
@@ -0,0 +1,184 @@
+
+VMX-pi provides I/O capabilities very similar to the roboRIO – making it useful for interfacing to external sensors (e.g., the Sparkfun `QWIIC Connect <https://www.sparkfun.com/qwiic>`_ family of I2C sensors), as well as the roboRIO CAN bus. On a FRC robot, these features can be integrated into your robot application (by sending information from VMX-pi to the roboRIO over the network via NetworkTables).  VMX-pi can also be used to unobtrusively monitor the behavior of a FRC robot for debugging and analyzing robot performance.
+
+Motion Processing Overview
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+VMX-pi includes a navX-Technology Inertial Measurement Unit (IMU), and includes 6 sensors which measure inertial motion:  3 accelerometers measuring acceleration (in units of `Standard Gravity <https://en.wikipedia.org/wiki/Standard_gravity>`_ [g]) and 3 gyroscopes measuring `Rotational Speed <https://en.wikipedia.org/wiki/Rotational_speed>`_ (in units of degrees per second).
+
+Additionally, through a process called “sensor fusion”, VMX-pi intelligently combines the 6-axis inertial sensing data to create a measurement of relative 3D orientation.
+
+.. image:: images/vmx-pi/yaw-pitch-roll.jpg
+
+IMUs are typically used to measure aircraft orientation, but are also very useful for controlling a robot.  IMUs measure rotation of an object around the Z-axis (yaw), the X-axis (pitch) and the Y-axis (roll).
+
+Pitch and roll angles are absolute (tied to the earth’s surface); 0 degrees means “flat” with respect to the earth.
+
+However, IMU Yaw angles are **relative** - not tied to any direction (like North on a compass).  Therefore, your robot application must decide where 0 degrees is.  Usually, FRC robots treat the “head” of the field (the direction driver’s face) as 0 degrees.
+
+For more information, please visit the `VMX-pi Terminology page <https://pdocs.kauailabs.com/vmx-pi/support/terminology/>`_.
+
+Digital Compass and Attitude/Heading Reference System (AHRS)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+VMX-pi’s navX-Technology also includes 3 magnetometer sensors, which measure magnetic fields (in units of Tesla).  By measuring the earth’s magnetic field, VMX-pi provides a digital compass – which is a different way of measuring the Z (“Yaw”) axis.
+
+.. image:: images/vmx-pi/compass.jpg
+
+And by intelligently fusing the digital compass with the gyroscope/accelerometer data, VMX-pi can create a measurement of absolute 3D orientation.
+
+.. note:: Earth’s magnetic field is actually very weak when compared to the magnetic field generated by a nearby motor; for this reason it can be difficult to get accurate digital compass readings on a FRC robot.  For this reason, using the VMX-pi AHRS is an advanced feature best suited for teams who have the time to learn about `how to calibrate the VMX-pi digital compass <https://vmx-pi.kauailabs.com/guidance/magnetometer-calibration/>`_ and also how to deal with magnetic disturbances.
```

Comment:
```suggestion
.. note:: Earth’s magnetic field is actually very weak when compared to the magnetic field generated by a nearby motor; for this reason it can be difficult to get accurate digital compass readings on a FRC robot.  For this reason, using the VMX-pi AHRS is an advanced feature best suited for teams who have the time to learn about `how to calibrate the VMX-pi digital compass <https://pdocs.kauailabs.com/vmx-pi/guidance/magnetometer-calibration/>`_ and also how to deal with magnetic disturbances.
```